core/block_svg.js

```diff
@@ -965,6 +977,11 @@ Blockly.BlockSvg.prototype.onMouseMove_ = function(e) {
       }
       this.setDragging_(true);
       this.moveToDragSurface_();
+
+
+      if (!Blockly.draggedConnectionManager_) {
```

Comment:
I think you should create a single DragManager when Blockly is instantiated and have it register an onMouseMove function that decides when to create a DragGesture or PendingDrag object that stores the drag information. BlockSVG should only have an onMouseDown that reports the down to the DragManager so it can keep track of which block was touched first.